### PR TITLE
bpo-29524: Add Objects/call.c file

### PR DESCRIPTION
**Pull request by vstinner**
* Move all functions to call objects in a new Objects/call.c file.
* Rename fast_function() to _PyFunction_FastCallKeywords(). 
* Copy null_error() from Objects/abstract.c
* Inline type_error() in call.c to not have to copy it, it was only   called once.
* Export _PyEval_EvalCodeWithName() since it is now called   from call.c.

http://bugs.python.org/issue29524

### CODE REVIEW

File: Python/ceval.c [hunk 1]
```diff
@@ -4887,205 +4850,6 @@ call_function(PyObject ***pp_stack, Py_ssize_t oparg, PyObject *kwnames)
...
 }
 
-/* The fast_function() function optimize calls for which no argument
-   tuple is necessary; the objects are passed directly from the stack.
-   For the simplest case -- a function that takes only positional
-   arguments and is called with only positional arguments -- it
-   inlines the most primitive frame setup code from
-   PyEval_EvalCodeEx(), which vastly reduces the checks that must be
-   done before evaluating the frame.
-*/
```
Comment (methane) [inline Python/ceval.c:4897]: This comment block should be moved to `function_code_fastcall()`.

**Review by methane — CHANGES_REQUESTED** [review]

**Comment by vstinner** [comment]
Oh in fact it was a deliberate choice to remove the comment. Python 3.7 now
has a wide range of "fast call" functions which are designed for best
performance. Do you think that it's worth it to keep the comment?

**Review by methane — APPROVED** [review]
I just checked all code is moved as-is.
I agree the comment block is not so important.

**Comment by vstinner** [comment]
Thank you for the review.

**Comment by jdemeyer** [comment]
Follow-up at #7909

**Pull request merged**